engine/access/rest/converter.go

```diff
@@ -3,21 +3,186 @@ package rest
+	address, err := toAddress(key.Address)
+	if err != nil {
+		return flow.ProposalKey{}, err
+	}
+
+	return flow.ProposalKey{
+		Address:        address,
+		KeyIndex:       uint64(key.KeyIndex),
+		SequenceNumber: uint64(key.SequenceNumber),
+	}, nil
+}
+
+func toSignature(signature *generated.TransactionSignature) (flow.TransactionSignature, error) {
+	address, err := toAddress(signature.Address)
+	if err != nil {
+		return flow.TransactionSignature{}, err
+	}
+
+	return flow.TransactionSignature{
+		Address:     address,
+		SignerIndex: int(signature.SignerIndex),
+		KeyIndex:    uint64(signature.KeyIndex),
+		Signature:   []byte(signature.Signature),
+	}, nil
+}
+
+func toSignatures(sigs []generated.TransactionSignature) ([]flow.TransactionSignature, error) {
+	signatures := make([]flow.TransactionSignature, len(sigs))
+	for _, sig := range sigs {
+		signature, err := toSignature(&sig)
+		if err != nil {
+			return nil, err
+		}
+
+		signatures = append(signatures, signature)
+	}
+
+	return signatures, nil
+}
+
+func toTransaction(tx *generated.TransactionsBody) (flow.TransactionBody, error) {
+	args := make([][]byte, len(tx.Arguments))
+	for _, arg := range tx.Arguments {
+		// todo validate
+		args = append(args, []byte(arg))
+	}
+
+	proposal, err := toProposalKey(tx.ProposalKey)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	payer, err := toAddress(tx.Payer)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	auths := make([]flow.Address, len(tx.Authorizers))
+	for _, auth := range tx.Authorizers {
+		a, err := toAddress(auth)
+		if err != nil {
+			return flow.TransactionBody{}, err
+		}
+
+		auths = append(auths, a)
+	}
+
+	payloadSigs, err := toSignatures(tx.PayloadSignatures)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	envelopeSigs, err := toSignatures(tx.EnvelopeSignatures)
+	if err != nil {
+		return flow.TransactionBody{}, err
+	}
+
+	return flow.TransactionBody{
+		ReferenceBlockID:   flow.Identifier{},
+		Script:             []byte(tx.Script),
+		Arguments:          args,
+		GasLimit:           uint64(tx.GasLimit),
+		ProposalKey:        proposal,
+		Payer:              payer,
+		Authorizers:        auths,
+		PayloadSignatures:  payloadSigs,
+		EnvelopeSignatures: envelopeSigs,
+	}, nil
+}
+
+// Response section - converting flow models to response models.
+
+func proposalKeyResponse(key *flow.ProposalKey) *generated.ProposalKey {
+	return &generated.ProposalKey{
+		Address:        key.Address.String(),
+		KeyIndex:       int32(key.KeyIndex),
+		SequenceNumber: int32(key.SequenceNumber),
+	}
+}
+
+func transactionSignatureResponse(signatures []flow.TransactionSignature) []generated.TransactionSignature {
+	sigs := make([]generated.TransactionSignature, len(signatures))
+	for _, sig := range signatures {
+		sigs = append(sigs,
+			generated.TransactionSignature{
+				Address:     sig.Address.String(),
+				SignerIndex: int32(sig.SignerIndex),
+				KeyIndex:    int32(sig.KeyIndex),
```

Comment:
this is a `uint64` in `flow.TransactionSignature`. any chance it overflows?

Reply:
actually keyIndex is the number of key that needs to be used - so I doubt if it will overflow since I would think the signer would have more than 2^32 keys. That being said, JSON doesn't support 64 bit integer data type so we will have to call this out somewhere in the swagger API.

I the protobug Access API also only supports the index as 32 bits - https://github.com/onflow/flow/blob/master/protobuf/flow/entities/transaction.proto#L26.

Reply:
@sideninja make sense! may i tackle this separately? - https://github.com/dapperlabs/flow-go/issues/6021. Realized this will need some unit tests of its own.

Reply:
for sure